virttest/qemu_devices/qdevices.py

```diff
@@ -1275,6 +1307,79 @@ def verify_unplug(self, out, monitor):
         return not self._is_attached_to_qemu(monitor)
 
 
+class QThrottleGroup(QObject):
+    """
+    throttle-group object.
+    attributes of throttle-group
+    """
+
+    KEY_MAP_TABLE = {"bps-read-max-length": "x-bps-read-max-length",
+                     "iops-read-max-length": "x-iops-read-max-length",
+                     "bps-read-max": "x-bps-read-max",
+                     "bps-total": "x-bps-total",
+                     "iops-total-max-length": "x-iops-total-max-length",
+                     "iops-total": "x-iops-total",
+                     "iops-write-max": "x-iops-write-max",
+                     "bps-write": "x-bps-write",
+                     "bps-total-max": "x-bps-total-max",
+                     "bps-write-max": "x-bps-write-max",
+                     "iops-size": "x-iops-size",
+                     "iops-read": "x-iops-read",
+                     "iops-write-max-length": "x-iops-write-max-length",
+                     "iops-write": "x-iops-write",
+                     "bps-total-max-length": "x-bps-total-max-length",
+                     "iops-read-max": "x-iops-read-max",
+                     "bps-read": "x-bps-read",
+                     "bps-write-max-length": "x-bps-write-max-length",
+                     "iops-total-max": "x-iops-total-max"}
+
+    def __init__(self, throttle_group_id, props):
+        self.raw_limits = json.loads(props) if isinstance(props,
+                                                          str) else props.copy()
+
+        params = QThrottleGroup._map_key(self.raw_limits)
+        params["id"] = throttle_group_id
+        kwargs = dict(backend="throttle-group", params=params)
+        super(QThrottleGroup, self).__init__(**kwargs)
+        self.throttle_group_bus = QThrottleGroupBus(throttle_group_id)
+        self.add_child_bus(self.throttle_group_bus)
+        self.set_aid(throttle_group_id)
+        self.hook_drive_bus = None
+
+    @staticmethod
+    def _check_key(params):
+        return set(params.keys()) <= set(QThrottleGroup.KEY_MAP_TABLE.keys())
```

Comment:
I think we don't need to check whether the param user passed is supported, let VM raise an error if the user passed the wrong throttling options in cfg. We default think that the user should configure the correct throttling options in cfg.

Reply:
ok